public/app/features/explore/QueryLibrary/QueryTemplatesList.tsx

```diff
@@ -1,14 +1,101 @@
-import { EmptyState, Spinner } from '@grafana/ui';
+import { css } from '@emotion/css';
+import { uniqBy } from 'lodash';
+import { useEffect, useMemo, useState } from 'react';
+
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { EmptyState, FilterInput, InlineLabel, MultiSelect, Spinner, useStyles2, Stack } from '@grafana/ui';
+import { t, Trans } from 'app/core/internationalization';
+import { createQueryText } from 'app/core/utils/richHistory';
 import { useAllQueryTemplatesQuery } from 'app/features/query-library';
 import { QueryTemplate } from 'app/features/query-library/types';
 
 import { getDatasourceSrv } from '../../plugins/datasource_srv';
 
+import { QueryLibraryProps } from './QueryLibrary';
 import QueryTemplatesTable from './QueryTemplatesTable';
 import { QueryTemplateRow } from './QueryTemplatesTable/types';
+import { searchQueryLibrary } from './utils/search';
+
+interface QueryTemplatesListProps extends QueryLibraryProps {}
 
-export function QueryTemplatesList() {
+export function QueryTemplatesList(props: QueryTemplatesListProps) {
   const { data, isLoading, error } = useAllQueryTemplatesQuery();
```

Comment:
Is there no part of the search that can be done on the backend? I would assume that at least description and query text filter can be easily done in the API.

Reply:
as we discussed, this is not currently possible because we can't search or filter the spec fields